crypto/encode_decode/decoder_lib.c

```diff
@@ -256,7 +256,7 @@ int OSSL_DECODER_CTX_add_extra(OSSL_DECODER_CTX *ctx,
              * on top of this one, so we don't.
              */
             if (ctx->start_input_type != NULL
-                && strcasecmp(ctx->start_input_type, input_type) != 0)
+                && strcasecmp(ctx->start_input_type, input_type) == 0)
```

Comment:
Wait what???

Reply:
Check the comment directly above this. We need to check whether the input types are equal, and not add more implementations if they are (hence we just "continue"). The logic was doing the reverse of what the comment said (only adding more implementations if the input types are the same). Without this change attempting to decode a PEM file by explicitly stating the "PEM"  input type fails to work.

Reply:
Ah, sorry, I was thinking of the similar condition down in `decoder_process` (line 501).
So, er, never mind.  Good catch!